crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/MeansAssessmentService.java

```diff
@@ -143,27 +144,11 @@ BigDecimal calculateSummariesTotal(final MeansAssessmentRequestDTO requestDTO, f
         return annualTotal;
     }
 
-    BigDecimal getDetailTotal(final ApiAssessmentDetail assessmentDetail, final boolean usePartner) {
-        BigDecimal detailTotal = BigDecimal.ZERO;
-
-        if (usePartner) {
-            BigDecimal partnerAmount = assessmentDetail.getPartnerAmount();
-            if (partnerAmount != null && !BigDecimal.ZERO.equals(partnerAmount)) {
-                Frequency partnerFrequency = assessmentDetail.getPartnerFrequency();
-                if (partnerFrequency != null) {
-                    detailTotal = detailTotal.add(partnerAmount.multiply(BigDecimal.valueOf(partnerFrequency.getWeighting())));
-                }
-            }
-        } else {
-            BigDecimal applicationAmount = assessmentDetail.getApplicantAmount();
-            if (applicationAmount != null && !BigDecimal.ZERO.equals(applicationAmount)) {
-                Frequency applicantFrequency = assessmentDetail.getApplicantFrequency();
-                if (applicantFrequency != null) {
-                    detailTotal = detailTotal.add(applicationAmount.multiply(BigDecimal.valueOf(applicantFrequency.getWeighting())));
-                }
-            }
-        }
-        return detailTotal;
+    BigDecimal calculateDetailTotal(BigDecimal amount, Frequency frequency) {
+        if (amount != null && frequency != null && !BigDecimal.ZERO.equals(amount)) {
+            return setStandardScale(setStandardScale(BigDecimal.valueOf(frequency.getWeighting()))
```

Comment:
I'm not sure we also need to set the scale on the frequency weighting.

Reply:
Setting a scale on the weighting as well result in 4 decimal points after the multiply, hence I removed it.